src/tests/SearchView.test.jsx

```diff
@@ -377,13 +421,22 @@ describe('Search View', () => {
     await user.type(searchInput, 'spamspamspamand@eggs.');
     await user.clear(searchInput);
     await user.type(searchInput, 'iamalmostlongenoughtobeahashbutnotquite');
-    await user.clear(searchInput);
+    jest.runOnlyPendingTimers();
 
     // fetch should only be called when selecting a repository
     expect(spyOnFetch).toHaveBeenCalledTimes(1);
     expect(spyOnFetch).toHaveBeenCalledWith(
       'https://treeherder.mozilla.org/api/project/autoland/push/',
     );
+
+    await waitFor(() =>
+      expect(
+        queryByText(
+          'Search must be a 12- or 40-character hash, or email address',
+        ),
+      ).toBeInTheDocument(),
+    );
```

Comment:
I believe `await screen.findByText('XXX')` would do the same and is more readable too.